test/pylib/scylla_cluster.py

```diff
@@ -402,17 +402,31 @@ async def start(self, api: ScyllaRESTAPIClient, expected_error: Optional[str] =
         sleep_interval = 0.1
         cql_up_state = CqlUpState.NOT_CONNECTED
 
+        def read_last_line(file_path: pathlib.Path):
+            max_line_bytes = 256
+            file_size = os.stat(file_path).st_size
+            with file_path.open('rb') as f:
+                f.seek(max(0, file_size - max_line_bytes), os.SEEK_SET)
+                line_bytes = f.read()
+            line_str = line_bytes.decode('utf-8', errors='ignore')
+            linesep = os.linesep
+            if line_str.endswith(linesep):
+                line_str = line_str[:-len(linesep)]
+            linesep_index = line_str.rfind(linesep)
+            if linesep_index != -1:
+                line_str = line_str[linesep_index + len(linesep):]
+            else:
+                line_str = '...' + line_str
```

Comment:
This gives kinda weird, unpredictable results
for the following files:
```python3
        "This is the first line.\nThis is the second line.\nThis is the third line.",
        "This is another file.\nIt has a few lines.\nThe last line is what we're interested in.",
        "This file has only one line.\n",
        "\n",
        "\n\n\n",
        ""
```

it gives the following corresponding results:
```
        'This is the third line.', 
        'The last line is what we\'re interested in.', 
        '...This file has only one line.', 
        '...', 
        '', 
        '...'
```

The appearance of `...` looks pretty random. IMO `...` should only be printed if we went over max_line_bytes.

Also 256 bytes may be low for Scylla log messages which are often quite long, I'd suggest to increase it to 512 bytes at least so we only cut messages that are very long compared to other Scylla messages.

Reply:
fixed